wrapper/fmriprep_docker.py

```diff
@@ -19,6 +19,7 @@
 import re
 import subprocess
 from warnings import warn
+from pathlib import Path
```

Comment:
This needs to be installable with zero dependencies on Python 2.

Reply:
oh. i forgot about that antiquated technology.

Reply:
Turns out this is a PITA without pathlib. See: https://stackoverflow.com/questions/3812849/how-to-check-whether-a-directory-is-a-sub-directory-of-another-directory/17624617#17624617

From what I can tell the only foolproof solutions are:
```
def os_path_split_asunder(path, debug=False):
    parts = []
    while True:
        newpath, tail = os.path.split(path)
        if debug: print repr(path), (newpath, tail)
        if newpath == path:
            assert not tail
            if path: parts.append(path)
            break
        parts.append(tail)
        path = newpath
    parts.reverse()
    return parts


def is_subdirectory(potential_subdirectory, expected_parent_directory):
    """
    Is the first argument a sub-directory of the second argument?
    """

    def _get_normalized_parts(path):
        return os_path_split_asunder(os.path.realpath(os.path.abspath(os.path.normpath(path))))

    # make absolute and handle symbolic links, split into components
    sub_parts = _get_normalized_parts(potential_subdirectory)
    parent_parts = _get_normalized_parts(expected_parent_directory)

    if len(parent_parts) > len(sub_parts):
        # a parent directory never has more path segments than its child
        return False

    # we expect the zip to end with the short path, which we know to be the parent
    return all(part1==part2 for part1, part2 in zip(sub_parts, parent_parts))
```

or maybe the following if we're okay with not being as thorough:

```
def is_in_directory(filepath, directory):
    return os.path.realpath(filepath).startswith(
        os.path.realpath(directory) + os.sep)
```

Reply:
I tested the latter a bit, and it seems to work even if the directory ends with a sep (e.g. : `/bids/dir/`). That was my main concern with that approach.

Reply:
Yeah, quick and dirty is fine. If someone wants to figure out a way to hack around it, they will, and I wish them well with their life choices.